pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java

```diff
@@ -26,12 +26,16 @@
  * The primary key of a record. Note that the value used in the primary key must be single-value.
  */
 public class PrimaryKey {
-  private final Object[] _values;
+  private Object[] _values;
```

Comment:
Let's not change this class. We don't want to replace the underlying value buffer